File: pyjl/tests_py2many-py2many/sealed.jl
using DataClass


@dataclass mutable struct Packet
val::Float64
_initvars = [_init=true, _repr=true, _eq=true, _order=false, _unsafe_hash=false, _frozen=false]
end


struct Register
PACKET::Packet
VALUE::Int64
end


function main()
a = VALUE(Register, 10)
@assert(is_value(a))
value(a);
b = PACKET(Register, Packet(1.3))
@assert(is_packet(b))
packet(b);
println("OK");
end

main()

File: tests/cases/sealed.py
#!/usr/bin/env python3

from dataclasses import dataclass

from adt import adt as sealed


@dataclass
class Packet:
    val: float


@sealed
class Register:
    PACKET: Packet
    VALUE: int


if __name__ == "__main__":
    a = Register.VALUE(10)
    assert a.is_value()
    a.value()
    # assert a.value() == 10
    b = Register.PACKET(Packet(1.3))
    assert b.is_packet()
    # assert b.packet().val == 1.3
    b.packet()
    print("OK")
